### PR TITLE
fix(lints): multiple linter rules #149

### LINKED ISSUE #149: Fix and enable linter rules
**Is your feature request related to a problem? Please describe.**

Yes, In flutter-lints dependency there are so many linter rules that I disabled in PR #148 

Now here are the checklists of the lint rules that are disabled (enabled rules are marked as checked) :

- [x] always_declare_return_types
- [x] avoid_print
- [x] avoid_unnecessary_containers
- [x] prefer_if_elements_to_conditional_expressions
- [x] prefer_const_constructors
- [x] prefer_const_declarations
- [x] unnecessary_string_interpolations
- [x] sized_box_for_whitespace
- [x] prefer_single_quotes
- [x] prefer_const_constructors_in_immutables
- [x] use_key_in_widget_constructors
- [x] prefer_function_declarations_over_variables
- [x] unnecessary_string_escapes
- [x] avoid_function_literals_in_foreach_calls
- [x] prefer_interpolation_to_compose_strings
- [ ] unnecessary_parenthesis
- [x] sort_constructors_first
- [x] always_put_required_named_parameters_first
- [ ] avoid_classes_with_only_static_members
- [ ] avoid_redundant_argument_values
- [x] join_return_with_assignment
- [ ] non_constant_identifier_names
- [ ] constant_identifier_names

### PR DESCRIPTION
**Pull request by Aadeesh11**
Fixes #149 

Describe the changes you have made in this PR -
Enabled linter:

1. non_constant_identifier_names
2. unnecessary_parenthesis
3. avoid_redundant_argument_values

(in order of changes required for them, high to low)
Note: Please check Allow edits from maintainers. if you would like us to assist in the PR.

### CODE REVIEW

**Comment by manjotsidhu** [comment]
Thanks for contributing, I will review it soon.

File: lib/viewmodels/about/about_viewmodel.dart [hunk 1]
```diff
@@ -7,7 +7,7 @@ import 'package:mobile_app/viewmodels/base_viewmodel.dart';
 
 class AboutViewModel extends BaseModel {
   // ViewState Keys
-  final String FETCH_CONTRIBUTORS = 'fetch_contributors';
```
Comment (manjotsidhu) [inline lib/viewmodels/about/about_viewmodel.dart:10]: I really liked the `ALL_CAPS_WITH_UNDERSCORES` naming convention but dart linter updated it to make lowerCamelCase. Though the dart rule also says, if we are already using ALL_CAPS_WITH_UNDERSCORES we can continue to use them but need to ignore this in linting. Refer: https://dart-lang.github.io/linter/lints/constant_identifier_names.html

This is really controversial decision and I will leave it to @Nitish145

Comment (Aadeesh11) [reply lib/viewmodels/about/about_viewmodel.dart:10]: Can we request a review? 😅

Comment (tachyons) [reply lib/viewmodels/about/about_viewmodel.dart:10]: Yes, changing this format doesn't add any extra value to the codebase. So it is better to keep this rule disabled

Comment (manjotsidhu) [reply lib/viewmodels/about/about_viewmodel.dart:10]: Sure, @Aadeesh11 you can disable `constant_identifier_names` and retain the ALL_CAPS_UNDERSCORE pattern wherever used.

Comment (Aadeesh11) [reply lib/viewmodels/about/about_viewmodel.dart:10]: @manjotsidhu done, but had to create a new PR. Please review that.

**Pull request closed without merging**